tests/backup/backup_test.go

```diff
@@ -60,14 +55,125 @@ func TearDownBackupRestore(bkpNamespaces []string, restoreNamespaces []string) {
+		defer dash.TestCaseEnd()
+		log.Infof(" No cleanup required for this testcase")
+	})
+})
+
+//This testcase verifies basic backup rule,backup location, cloud setting
+var _ = Describe("{BasicBackupCreateWithRules}", func() {
+	var contexts []*scheduler.Context
+	var appContexts []*scheduler.Context
+	JustBeforeEach(func() {
+		dash.Info("Deploy applications")
+		dash.TestCaseBegin("Backup: BasicBackupCreateWithRules", "Creating backup with Rules", "", nil)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+		}
+	})
+	It("Baic Backup Creation", func() {
+		var (
+			ps               = make(map[string]map[string]string)
+			create_pre_rule  = false
+			create_post_rule = false
+			app_list         = Inst().AppList
+		)
+		Step("Validate applications and get their labels", func() {
+			ValidateApplications(contexts)
+			logrus.Infof("Create list of pod selector for the apps deployed")
+			for _, ctx := range appContexts {
+				for _, specObj := range ctx.App.SpecList {
+					if obj, ok := specObj.(*appsapi.Deployment); ok {
+						if contains(app_list, obj.Name) == true {
+							ps[obj.Name] = obj.Spec.Template.Labels
+						}
+					} else if obj, ok := specObj.(*appsapi.StatefulSet); ok {
+						if contains(app_list, obj.Name) == true {
+							ps[obj.Name] = obj.Spec.Template.Labels
+						}
+					}
+				}
+			}
+		})
+		Step("Creating rules for backup", func() {
+			pre_ps_list := []map[string]string{}
+			pre_actions := []string{}
+			pre_container_list := []string{}
+			pre_background := []bool{}
+			pre_run_in_single_pod := []bool{}
+			dash.Info(" Creating pre rule for deployed apps")
+			for i := 0; i < len(app_list); i++ {
+				if app_list[i] == "cassandra" || app_list[i] == "postgres" {
+					create_pre_rule = true
+					pre_ps_list = append(pre_ps_list, ps[app_list[i]])
+					pre_actions = append(pre_actions, pre_action_list[app_list[i]])
+					pre_background = append(pre_background, background_check_dict[app_list[i]])
+					pre_run_in_single_pod = append(pre_run_in_single_pod, run_in_single_pod_dict[app_list[i]])
+					// Here user has to set env for each app container if required in the format container<app name> eg: containersql
+					container_name := fmt.Sprintf("%s%s", "container", app_list[i])
+					fmt.Println(" Conainer name is ", container_name)
+					pre_container_list = append(pre_container_list, os.Getenv(container_name))
+				}
+			}
+			if create_pre_rule == true {
+				pre_rule_status := createRuleForBackup("backup-pre-rule", "default", pre_ps_list, pre_actions,
+					pre_background, pre_run_in_single_pod, pre_container_list)
+				fmt.Println(pre_rule_status)
+				dash.VerifyFatal(pre_rule_status, true, fmt.Sprintf("Verifying pre rule for backup"))
+			}
+			dash.Info(" Creating post rule for deployed apps")
```

Comment:
Unnecessary space at the start of the log line.

Reply:
done!!